apps/api/src/controllers/user.controller.ts

```diff
@@ -167,18 +167,7 @@ export function makeUserController(services: ServiceRegistry) {
         // Return the created agent with API key (user needs this for distribution)
         res.status(201).json({
           success: true,
-          agent: {
-            id: agent.id,
-            ownerId: agent.ownerId,
-            name: agent.name,
-            description: agent.description,
-            imageUrl: agent.imageUrl,
-            metadata: agent.metadata,
-            apiKey: agent.apiKey, // Include API key for user to use
```

Comment:
ah, this is the one place where we _do_ want to expose the `apiKey`; else, when a user creates an agent, they wont be able to get the agent's API key in the response.

since `sanitizeAgent` sanitizes some of the other fields, maybe it can take an optional to `includeApiKey: boolean = false` or something? or just add a separate method.

Reply:
Oh yeah, duh.

Reply:
partially my fault...the user tests aren't comprehensive enough, so this got missed in logic checks.